DucklingHTTPExtractor with defined request timeout

vishnuvrpriya opened the pull request:
**Proposed changes**:
- Fixed the bug being discussed under #1695 
- Added a timeout of a default of 3 seconds to t.he Duckling HTTP Request Invocation and made it a configurable component parameter.
- Updated the docs to reflect the new timeout parameter.
- Ran a few tests to ascertain the fix.Attached a screenshot (below) of the test that fulfills the fix.
![DucklingHTTPTimeout](https://user-images.githubusercontent.com/32483075/65392065-1477c980-dd8e-11e9-818e-09687a6e41a9.png)
 

**Status (please check what you already did)**:
- [x] made PR ready for code review 
- [ ] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/RasaHQ/rasa?pullRequest=4507) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/RasaHQ/rasa?pullRequest=4507) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: Rangachari Vangipuram<br/>:x: vishnuvrpriya<br/><hr/>**Rangachari Vangipuram** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/RasaHQ/rasa?pullRequest=4507) it.</sub>

msamogh commented:
Thanks for the PR, we'll give it a review as soon as possible.

imLew commented on rasa/nlu/extractors/duckling_http_extractor.py:
```diff
@@ -89,6 +92,10 @@ def _locale(self):
         return self.component_config.get("locale")
 
+
+    def _timeout(self):
```
I don't think it's necessary to create a method for this, as it' only getting the value from the dict.

vishnuvrpriya replied:
Addressed this review comment.

imLew commented on rasa/nlu/extractors/duckling_http_extractor.py:
```diff
@@ -113,7 +120,7 @@ def _duckling_parse(self, text, reference_time):
             }
             response = requests.post(
-                self._url() + "/parse", data=payload, headers=headers
+                self._url() + "/parse", data=payload, headers=headers,timeout=self._timeout()
```
```suggestion
                self._url() + "/parse", data=payload, headers=headers, timeout=self.component_config.get("timeout")
```

You can simply get the value from the dict here.

vishnuvrpriya replied:
Addressed this review comment.

vishnuvrpriya commented:
Hi @imLew ,Thanks for the valuable comments.I will address them and get back to you!

vishnuvrpriya commented:
Hi @imLew , I have addressed the review comments that you suggested.I have also formatted the code using black and have successfully finished signing the CLA.

imLew commented:
@vishnuvrpriya thanks!
Just one more thing: it seems like github does not know that your local git user is the same as your github user, that's why there are two names that needs to sign the CLA. Unfortunately we cannot merge this PR unless all contributors have signed the CLA.
I've done bit of searching and maybe the answers to https://webapps.stackexchange.com/questions/20309/how-to-claim-contributions-on-github can help you claim the commits you made locally.

vishnuvrpriya commented:
Hi @imLew . I am unable to solve the CLA issue and so I am trying to close and reopen the PR. Please bear with me for the inconvenience.

vishnuvrpriya commented:
Hi @imLew , I am unable to solve the CLA issue and I am trying to close and reopen the PR.Please bear with me for the inconvenience.

imLew commented:
@vishnuvrpriya am I right in assuming the git user that did not sign the CLA is your local user.name?
If so, does it have an email address associated with it? If yes you should be able to add that email address to your github account. Once it's added the CLA should either be signed or you should be able to sign it both users.
If not I'm not sure if you can change this PR. You might however be able to somehow change it locally (e.g. adding an email address to it) and create another PR.

imLew commented:
@vishnuvrpriya, I asked around and it seems like you will have to create a new PR if you cannot add the email to your github account.
To get the new PR to associate the commits with your github account you can do a soft reset or you can try https://github.com/jayphelps/git-blame-someone-else (note that I haven't tried either of these methods myself). Before that of course make sure that your local git users email address is the same as the one linked to your github account.

vishnupriyavr commented:
Hi @imLew ,Thanks for the guidance.I actually had to create a new user and a new PR #4535 to successfully complete the CLA.I request you to review the PR and help in merging it.I will also be closing this PR to avoid any confusion.
